<a href="https://colab.research.google.com/github/AkshatkMalik/Python_Programming/blob/main/device_satisfaction_chisquare.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**ASSIGNMENT-4**

**Statement of Hypothesis:**

Null Hypothesis ($H_0$): There is no relation between the kind of smart home device bought by customers (Smart Thermostats vs. Smart Lights) and their satisfaction level (these are independent variables).

Alternative Hypothesis ($H_a$): There exists a significant relation between the kind of smart home device bought by customers and their satisfaction level (these are dependent variables).

In [2]:
import numpy as np
from scipy import stats

# 1. CreatING the contingency table from the data provided
# Rows: [Very-Satisfied, Satisfied, Neutral, Unsatisfied, Very Unsatisfied]
# Columns: [Smart Thermostat, Smart Light]
observed_data = np.array([
    [50, 70],   # Very-Satisfied
    [80, 100],  # Satisfied
    [60, 90],   # Neutral
    [30, 50],   # Unsatisfied
    [20, 50]    # Very Unsatisfied
])

# 2. Run the Chi-Square Test for Independence
chi2_stat, p_value, dof, expected_frequencies = stats.chi2_contingency(observed_data)

# 3. DeterminING the Critical Value for alpha = 0.05
alpha = 0.05
critical_value = stats.chi2.ppf(1 - alpha, dof)

# 4. Printing out all required results cleanly
print(" CHI-SQUARE TEST RESULTS ")
print(f"Chi-Square Statistic: {chi2_stat:.4f}")
print(f"Degrees of Freedom (dof): {dof}")
print(f"Critical Value (alpha = 0.05): {critical_value:.4f}")
print(f"P-Value: {p_value:.4f}")

# 5. Making an automated decision
print("\n DECISION ")
if chi2_stat > critical_value:
    print("Result: Reject the Null Hypothesis (H0). There is a significant association.")
else:
    print("Result: Fail to Reject the Null Hypothesis (H0). No significant association found.")

 CHI-SQUARE TEST RESULTS 
Chi-Square Statistic: 5.6382
Degrees of Freedom (dof): 4
Critical Value (alpha = 0.05): 9.4877
P-Value: 0.2278

 DECISION 
Result: Fail to Reject the Null Hypothesis (H0). No significant association found.


1. The Core Formulas Chi-Square Statistic:


  $\chi^2 =\sum \frac{(O - E)^2}{E}$Expected Frequency ($E$): $E = \frac{(\text{Row Total}) \times (\text{Column Total})}{\text{Grand Total }(N=600)}$

2. Calculating Expected Frequencies ($E$) Examples


  Using the formula for each category (Thermostat column total = 240, Light column total = 360):

  Very Satisfied: Thermostat = $\frac{120 \times 240}{600} = 48.0$ | Light = $\frac{120 \times 360}{600} = 72.0$

  Satisfied: Thermostat = $\frac{180 \times 240}{600} = 72.0$ | Light = $\frac{180 \times 360}{600} = 108.0$

  (Neutral, Unsatisfied, and Very Unsatisfied follow the exact same proportional logic based on their respective row totals).

3. Computing $\frac{(O - E)^2}{E}$ and Summing


  For each cell, we take the squared difference between Observed ($O$) and Expected ($E$), divided by $E$:

  Very Satisfied: $\frac{(50-48)^2}{48} + \frac{(70-72)^2}{72} = 0.0833 + 0.0556 = 0.1389$

  Satisfied: $\frac{(80-72)^2}{72} + \frac{(100-108)^2}{108} = 0.8889 + 0.5926 = 1.4815$

  Neutral: $\frac{(60-60)^2}{60} + \frac{(90-90)^2}{90} = 0.0000$

  Unsatisfied: $\frac{(30-32)^2}{32} + \frac{(50-48)^2}{48} = 0.1250 + 0.0833 = 0.2083$

  Very Unsatisfied: $\frac{(20-28)^2}{28} + \frac{(50-42)^2}{42} = 2.2857 + 1.5238 = 3.8095$

4. Final Chi-Square Statistic



  $$\chi^2 = 0.1389 + 1.4815 + 0.0000 + 0.2083 + 3.8095 = \mathbf{5.6382}$$


**Critical Value & Decision Rule:**

Significance Level ($\alpha$): $0.05$

Critical Value: $9.4877$

Comparison: Since the calculated Chi-Square statistic ($5.6382$) is less than the critical value ($9.4877$)—and the p-value ($0.2278$) is greater than $0.05$—we fail to reject the null hypothesis ($H_0$).

**Conclusion and Business Implication:**

Conclusion on Statistics: There is no statistically significant correlation between the type of smart home device being bought and the satisfaction rating of customers.

Implication for Mizzare Corporation: The level of satisfaction of customers is independent of the product choice. The quality control and customer experience teams of Mizzare need to investigate other factors such as price and software and hardware reliability instead of assuming that any dissatisfaction is due to selection of the device type.